lib/expression/callbacks/standard.ex

```diff
@@ -116,6 +116,92 @@ defmodule Expression.Callbacks.Standard do
+                  expression: ~s[SWITCH(5, 1, "Sunday", 2, "Monday", 3, "Tuesday", "No match")],
+                  result: "No match"
+  def switch_vargs(ctx, arguments) do
+    [key | options] = eval_args!(arguments, ctx)
+    optional = if rem(length(options), 2) == 1, do: List.last(options), else: nil
+
+    options
+    |> Enum.chunk_every(2, 2, :discard)
+    |> Map.new(fn [a, b] -> {a, b} end)
+    |> Map.get(key, optional)
+  end
+
+  @expression_doc doc:
+                    "The ROUND function rounds a number to a specified number of digits. For example, if cell A1 contains 23.7825, and you want to round that value to two decimal places you can do ROUND(23.7825, 2)",
+                  expression: "ROUND(23.7825, 2)",
+                  result: "23.78"
+  @expression_doc doc:
+                    "The ROUND function rounds a number to a specified number of digits. For example, if cell A1 contains 23.7825, and you want to round that value to zero decimal places you can do ROUND(23.7825)",
+                  expression: "ROUND(23.7825)",
+                  result: "24"
+  def round(ctx, value) do
+    [value] = eval_args!([value], ctx)
+
+    value
+    |> Decimal.from_float()
+    |> Decimal.round(0)
+    |> Decimal.to_string(:normal)
+  end
+
+  def round(ctx, value, places) do
+    [value, places] = eval_args!([value, places], ctx)
+
+    value
+    |> Decimal.from_float()
+    |> Decimal.round(places)
+    |> Decimal.to_string(:normal)
+  end
+
+  @doc """
+  MID extracts part of a string, starting at a specified position and for a specified length.
+
+  It correctly handles Unicode characters. For example, taking the first three characters from "héllo" returns "hél".
+
+  If the starting position is beyond the string length, it returns an empty string.
+
+  Implementation based on https://support.microsoft.com/en-us/office/mid-function-2eba57be-0c05-4bdc-bf81-5ecf4421eb8a
+  """
+  @expression_doc doc:
+                    "MID returns a specific number of characters from a text string, starting at the position you specify, based on the number of characters you specify.",
+                  expression: ~s[MID("Fluid", 1, 5)],
+                  result: "Fluid"
+
+  @expression_doc expression: ~s[MID("Fluid Flow", 7, 20)],
+                  result: "Flow"
+
+  @expression_doc expression: ~s[MID("Fluid Flow", 20, 5)],
+                  result: ""
+
+  def mid(ctx, text, start_num, num_chars) do
+    [text, start_num, num_chars] = eval_args!([text, start_num, num_chars], ctx)
+    String.slice(to_string(text), start_num - 1, num_chars)
+  end
+
+  @expression_doc doc:
+                    "Convert a date into any strftime format (ref: https://man7.org/linux/man-pages/man3/strftime.3.html)",
+                  expression: ~s[text(datevalue(date(2022, 09, 14)), "%m/%d/%Y")],
+                  result: "09/14/2022"
+  def text(ctx, value, format) do
```

Comment:
How is this different from supplying the format directly to `datevalue` here? That function accepts a string formatting argument already: https://github.com/turnhub/expression/blob/81ea56047cd9fdade0d5daee9c89bd0e450fc461/lib/expression/callbacks/standard.ex#L119-L149

<img width="981" alt="Screenshot 2024-08-27 at 11 18 07" src="https://github.com/user-attachments/assets/1ce42cf3-e132-42d4-84e9-bece835c2aba">

I can see the `text()` function name here being ambiguous and likely a source of confusion for folks given this seems like a generic function but is only really useful for date / datetime things.

That said, if you're wanting to use this in your own code, you could perhaps include it in a custom callbacks implementation ? The implementation isn't ideal but here's an example test case that does exactly that: https://github.com/turnhub/expression/blob/develop/test/expression_custom_callbacks_test.exs

Reply:
Reading https://support.microsoft.com/en-us/office/text-function-20d5ac4d-7b94-49fd-bb38-93d29371225c I understand why it's called `text()`, I guess it's a generic formatting function.

However, if we want to implement this we'll want to make sure to support other types of values as well. Until that is ready I would suggest keeping this one in a custom callbacks implementation?

Reply:
I would say we could remove this from this MR and leave it as custom callback, sounds good to me!